diff --git a/term_merge.module b/term_merge.module
index d78d3fb..64e7656 100644
--- a/term_merge.module
+++ b/term_merge.module
@@ -447,7 +447,7 @@ function _term_merge_insert_field_values($update_data, &$context) {
       if ($value['tid'] == $update_data_arr['source_tid']) {
 
         if (!$target_merged) {
-          $field_values[$i]['tid'] = $update_data_arr['dest_tid']->tid;
+          $field_values[$i]['tid'] = $update_data_arr['dest_tid'];
           $target_merged = TRUE;
         }
         // Any more occurrences of the source tid will simply be removed.
@@ -458,7 +458,7 @@ function _term_merge_insert_field_values($update_data, &$context) {
 
         // If target tid is one of the source tids, remove it from the array
         // of terms that will be removed at the end of the process.
-        if($update_data_arr['source_tid']== $update_data_arr['dest_tid']->tid) {
+        if($update_data_arr['source_tid']== $update_data_arr['dest_tid']) {
           foreach ($context['results']['merged_terms_delete'] as $i => $merged_term) {
             if ($merged_term == $update_data_arr['source_tid']) {
               array_splice($context['results']['merged_terms_delete'], $i, 1);
@@ -467,7 +467,7 @@ function _term_merge_insert_field_values($update_data, &$context) {
         }
 
         $context['results'][] = 'Merged ' . $update_data_arr['source_tid'] . ' with '
-        . $update_data_arr['dest_tid']->tid . ' in ' . $update_data_arr['field_name']
+        . $update_data_arr['dest_tid'] . ' in ' . $update_data_arr['field_name']
         . ' of ' . $entity_type . ' ' . $entity_id;
 
         break;
@@ -486,6 +486,7 @@ function _term_merge_insert_field_values($update_data, &$context) {
   if ($target_merged) {
     field_attach_presave($entity_type, $entity);
     field_attach_update($entity_type, $entity);
+    taxonomy_node_update($entity);
   }
 }
 
